Client/multiplayer_sa/CMultiplayerSA_DeviceSelection.cpp

```diff
@@ -0,0 +1,157 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        multiplayer_sa/CMultiplayerSA_DeviceSelection.cpp
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+#ifdef _WIN32
+    #ifndef NOMINMAX
+        #define NOMINMAX
+    #endif
+#endif
+
+// This is copied from SilentPatch:
+// https://github.com/CookiePLMonster/SilentPatch/blob/dev/SilentPatch/FriendlyMonitorNames.cpp
+std::map<std::string, std::string, std::less<>> GetFriendlyMonitorNamesForDevicePaths()
+{
+    std::map<std::string, std::string, std::less<>> monitorNames;
```

Comment:
unordered_map instead of map

Reply:
Pointless in this case, under normal conditions there will be 2-3 entries in the map. I don't think this will be higher than 6 ever. The overhead of unordered_map makes it worse choice in this case. 
As soon as someone wants to run MTA on the Las Vegas Sphere, I will gladly change it :D
![462555296_9454252167923288_2572623932954594248_n](https://github.com/user-attachments/assets/2216e996-6d0d-4c0f-8663-859ae3a16046)

Reply:
> Pointless in this case, under normal conditions there will be 2-3 entries in the map. I don't think this will be higher than 6 ever. The overhead of unordered_map makes it worse choice in this situation. As soon as someone wants to run MTA on the Las Vegas Sphere, I will gladly change it :D ![462555296_9454252167923288_2572623932954594248_n](https://private-user-images.githubusercontent.com/4185198/384704953-2216e996-6d0d-4c0f-8663-859ae3a16046.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyNTc3MDIsIm5iZiI6MTczMTI1NzQwMiwicGF0aCI6Ii80MTg1MTk4LzM4NDcwNDk1My0yMjE2ZTk5Ni02ZDBkLTRjMGYtODY2My04NTlhZTNhMTYwNDYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTBUMTY1MDAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjcyNTgyNDFhNzU2MmUzYWNhOGEzODQyMzIzNmUxNzRkMTMyZmM5NTAzN2EzMWZiN2JkYzE5MDcxNDgxZjUyYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.bS-TzVnviDZZU-rMUpWnYnAJ074WtAVEEaxbosD5EL4)

There is no overhead. In fact `map` is slower than `unordered_map` even when the amount of entries is small.
![image](https://github.com/user-attachments/assets/fecb05ad-a5b7-42df-8af7-31d6d0e355ed)
![image](https://github.com/user-attachments/assets/de860679-63ea-4b3a-bfab-48aecd4aa401)

Reply:
1. This is an absolute nitpick, doesn't affect performance/stability of the game at all, and is purely due to your personal style preferences.
2. Code guidelines (https://github.com/multitheftauto/mtasa-docs/blob/main/mtasa-blue/CODING_GUIDELINES.md) do not specify anything about using unordered_map over map
3. Minor stuff like this will only discourage people from contributing to the project, this is a non-topic in any normal scenario, and the code runs one time during startup for few items.
4. Your benchmark is also invalid, most of the benchmarks like these can be made to fit any thesis, it tests functions that are not even used in my code (Not really mine, it's Silents), when we replicate the actual logic that's happening suddenly map is very slightly ahead:
![obraz](https://github.com/user-attachments/assets/c5d655f1-302e-4d8d-8b05-a32a55ea27dd)

Reply:
> 1. This is an absolute nitpick, doesn't affect performance/stability of the game at all, and is purely due to your personal style preferences.
> 2. Code guidelines (https://github.com/multitheftauto/mtasa-docs/blob/main/mtasa-blue/CODING_GUIDELINES.md) do not specify anything about using unordered_map over map
> 3. Minor stuff like this will only discourage people from contributing to the project, this is a non-topic in any normal scenario, and the code runs one time during startup for few items.
> 4. Your benchmark is also invalid, most of the benchmarks like these can be made to fit any thesis, it tests functions that are not even used in my code (Not really mine, it's Silents), when we replicate the actual logic that's happening suddenly map is very slightly ahead:
>    ![obraz](https://private-user-images.githubusercontent.com/4185198/384719226-c5d655f1-302e-4d8d-8b05-a32a55ea27dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyNzEwOTAsIm5iZiI6MTczMTI3MDc5MCwicGF0aCI6Ii80MTg1MTk4LzM4NDcxOTIyNi1jNWQ2NTVmMS0zMDJlLTRkOGQtOGIwNS1hMzJhNTVlYTI3ZGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTBUMjAzMzEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWUxNzNhZThmMDQxODIzMzljY2U4ODllYmE5MGY3ODAwMWEzZWY3MzExMjhiNTZjMTllOGEzZDJlMTQwNzIzNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.45f1CisrOhxtXLqlenWnGIRiI-jTYkqI7kqaLKJgDRc)

1. This is not nitpicking. This is not a matter of personal style preference. This is like comparing `std::string` to `std::vector<char>`.
2. Guidelines dont dictate which code should be used when performance is at stake.
`map` **is** slower than `unordered_map` due to `map` sorting elements. `unordered_map` therefore in theory takes more memory than normal map but is faster in the end.
3. Reviewing code does not discourage people from contributing. Its weird you feel that way.
4. Your benchmark differs. One isnt equivalent to another. Why do you try to manipulate the results this way?

Reply:
1. In my opinion it is nitpicking.
2. There's no performance at stake, pretty much every single other function called in that scope (GetFriendlyMonitorNamesForDevicePaths) will take orders of magnitude longer than all the map/unordered_map operations there, it's literally few items added to map, and a single check to find matching item afterwards. This is purely style issue.
3. Reviewing code is completely okay, every single other comment in this PR has been fixed without any complaints, it's this one that's a problem to me.
4. It differs because that's the actual usage in the code that's part of this PR, if you want to fight about this then let's consider exactly the same handling code (it's true that i missed the fact that one uses [key] = value and second one used try_emplace):
![obraz](https://github.com/user-attachments/assets/27373201-e97c-4a9c-9f02-8791f4231198)
![obraz](https://github.com/user-attachments/assets/b044cdbf-2874-4f2a-a6ec-e6f0d7fb303c)

As i said, those micro-benchmarks are completely worthless, and you can adjust the code both ways to always prove your thesis.

Reply:
1. Nitpicking is only applicable for insignificant changes. If changing functionality (chanigng how X works -> changing functionality) and debating about performance is insignificant then every code review is nitpicking.
2. You dont seem to understand what `style issue` is. This is **not** a style manner. Even if one function takes longer than another there is no reason to bottleneck it and make it slower.
4. I did some more digging and it turns out only the initialization is slower for the unordered_map. Providing `std::less` to map has no effect because `map` already takes in `std::less` as its `Compare` template argument. Check the wiki
Benchmarks:
- [`unordered_map` vs `map<less<>>` with initialization](<https://quick-bench.com/q/d0-EC0XW0um4LwIvulwd22ly5H8>)
- [`unordered_map` vs `map` with initialization](<https://quick-bench.com/q/G8IVphVpAOlc6Tnd5zMdLeYE-fU>)
- [`unordered_map` vs `map<less<>>` without initialization](<https://quick-bench.com/q/7UVnBlPEcwoPBo8PKEpKqKsbkNY>)
- [`unordered_map` vs `map` without initialization](<https://quick-bench.com/q/9Ua2Y50K5898djUY1RkCPo4rb7U>)

As you can see only initialization takes more time, but is more performant in the end.
TLDR; `std::map<std::string, std::string, std::less<>>` resolves to `std::map<std::string, std::string>` due to the nature of `std::less`. So basically we are still comparing `std::map` vs `std::unordered_map`.

Reply:
Congratulations, You've spent more on this analysis than the code you're arguing about will spend searching for an entry in this 2-4 elements big map in the coming years. There's only 486 other usages of std::map that can be fixed in the codebase - hopefully one day soon.
Changed to unordered_map now - i really hope to see someone running MTA on a setup that uses hundreds of displays so they can take advantage of those few nanoseconds gained on startup :)

Reply:
If we stop caring for performance, we should stop caring about the crashes. And if we stop caring about the crashes we should stop caring at all. You can see where this is going.

Reply:
> Congratulations, You've spent more on this analysis than the code you're arguing about will spend searching for an entry in this 2-4 elements big map in the coming years. There's only 486 other usages of std::map that can be fixed in the codebase - hopefully one day soon. Changed to unordered_map now - i really hope to see someone running MTA on a setup that uses hundreds of displays so they can take advantage of those few nanoseconds gained on startup :)

XD

Tracer's mindset is If we can change it to be more performant, why wouldn't we? I gotta agree if it takes little effort to make a refactor from std map to std unordered map, for example.